great_expectations/expectations/expectation.py

```diff
@@ -1003,6 +1001,174 @@ def run_diagnostics(self, pretty_print=True):
 
         return report_obj
 
+    def generate_diagnostic_checklist(self) -> str:
+
+        diagnostics_report = self.run_diagnostics()
+
+        checks = []
+
+        # Check whether this Expectation has a library_metadata object
+        checks.append(
+            {
+                "message": "library_metadata object exists",
+                "passed": hasattr(self, "library_metadata")
+                and type(self.library_metadata) == dict
+                and set(self.library_metadata.keys())
+                == {"maturity", "package", "tags", "contributors"},
+            }
```

Comment:
What would that look like for users? Or are you thinking of this as a purely internal check?

Related: in a subsequent PR, I'd suggest turning `library_metadata` into a typed object, so that these checks can be enforced at instantiation, with sensible error meesages.

Reply:
I'm thinking of this as a purely internal thing - just a refactor of existing code. Not critical, but could make the code easier to update down the road.